src/pretix/base/migrations/0219_auto_20220706_0913.py

```diff
@@ -0,0 +1,38 @@
+# Generated by Django 3.2.12 on 2022-07-06 09:13
+
+import django.db.models.deletion
+import i18nfield.fields
+from django.db import migrations, models
+
+import pretix.base.models.base
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('pretixbase', '0218_checkinlist_addon_match'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='CustomerSSOProvider',
+            fields=[
+                ('id', models.BigAutoField(primary_key=True, serialize=False)),
+                ('name', i18nfield.fields.I18nCharField(max_length=200)),
+                ('is_active', models.BooleanField(default=True)),
+                ('button_label', i18nfield.fields.I18nCharField(max_length=200)),
+                ('method', models.CharField(max_length=190)),
+                ('configuration', models.JSONField()),
+                ('organizer', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='sso_providers', to='pretixbase.organizer')),
+            ],
+            options={
+                'abstract': False,
+            },
+            bases=(models.Model, pretix.base.models.base.LoggingMixin),
+        ),
+        migrations.AddField(
+            model_name='customer',
+            name='provider',
+            field=models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, related_name='customers', to='pretixbase.customerssoprovider'),
```

Comment:
Shouldn’t this be `SET_NULL` instead of `CASCADE`? Or do we want the customer account to be deleted if we remove the SSOProvider as without without the SSOProvider that specific customer could not login (although currently email+password would still work)?

Reply:
I think this should actually be PROTECT

Reply:
You are right! I thought that, if a SSOProvider is compromised, the organizer could simply remove it – but it is better to block delete, when that SSOProvider is in use and have the organizer set the SSOProvider’s `active` attribute to `False`.